src/olympia/addons/tests/test_tasks.py

```diff
@@ -0,0 +1,72 @@
+import mock
+import os
+import tempfile
+
+from django.conf import settings
+
+from PIL import Image
+
+from olympia.amo.tests import addon_factory, TestCase
+from olympia.amo.tests.test_helpers import get_image_path
+from olympia.addons.tasks import (
+    create_persona_preview_images, save_persona_image)
+
+
+class TestPersonaImageFunctions(TestCase):
+    @mock.patch('olympia.addons.tasks.pngcrush_image')
+    def test_create_persona_preview_image(self, pngcrush_image_mock):
+        addon = addon_factory()
+        addon.modified = self.days_ago(41)
+        # Given an image, a 680x100 and a 32x32 thumbnails need to be generated
+        # and processed with pngcrush.
+        expected_dst1 = tempfile.NamedTemporaryFile(
+            mode='r+w+b', suffix=".png", delete=False, dir=settings.TMP_PATH)
+        expected_dst2 = tempfile.NamedTemporaryFile(
+            mode='r+w+b', suffix=".png", delete=False, dir=settings.TMP_PATH)
+        create_persona_preview_images(
+            src=get_image_path('persona-header.jpg'),
+            full_dst=[expected_dst1.name, expected_dst2.name],
+            set_modified_on=[addon],
+        )
+        # pngcrush_image should not been called twice, once for each
+        # destination thumbnail.
+        assert pngcrush_image_mock.call_count == 2
+        assert pngcrush_image_mock.call_args_list[0][0][0] == (
+            expected_dst1.name)
+        assert pngcrush_image_mock.call_args_list[1][0][0] == (
+            expected_dst2.name)
```

Comment:
This is quite ugly, maybe you can utilize `assert_called_with` or `assert_has_calls` to make this easier to read?